### DISCUSSION TITLE
registration_fgr_based_on_feature_matching stuck

### DISCUSSION THREAD

**Discussion by eladlavi** [isl-org/Open3D#4423] [General]
Hi there. registration_fgr_based_on_feature_matching gets stuck for no obvious reason.

I am running it in a loop:
o3d.pipelines.registration.registration_fgr_based_on_feature_matching(
        source_down, target_down, source_fpfh, target_fpfh,
        o3d.pipelines.registration.FastGlobalRegistrationOption(
            maximum_correspondence_distance=distance_threshold,
            iteration_number=10))

I am trying to register a series of consecutive point clouds, not too big, about 50,000 points,
from a scanner, the scanner is in static environment, doesn't event move much so the point clouds are almost identical.
Using a strong PC. It succeeds multipe times (again for the very similar point clouds, only minor noise of difference), and then, at some iteration, it gets stuck (I can hear the PC fan going crazy).

I am using the registration_fgr_based_on_feature_matching to get an initial guess for the transformation and then I pass it to the ICP algorithm.

What can cause it to get internally in an infinite loop ? again, for the same point cloud data it succeed multiple times before this happens.
Perhaps there's no correct release of resources internally within the code ?

Thanks.

**Comment by michezio** [comment]
Hi I'm having the same issue with a similar setup, did you manage to solve it somehow?
As far as I can tell it has some kind of memory leak since when it gets stuck CPU load goes down and RAM suddendly ramps up to max

**Reply by eladlavi** [reply]
Hi. Unfortunately, NO.
So I abandoned the fast global registration and switched to RANSAC (the slow global registration).
I also think there is a bug that causes memory allocation instead of reusing and recycling memory objects.

**Reply by michezio** [reply]
I switched to RANSAC too, but I think it should be something worth of investigation, I'm opening an issue

**Comment by germanros1987** [comment]
@ssheorey can look into this, but we would need some pointclouds.

**Comment by germanros1987** [comment]
@ssheorey can look into this, but we would need some pointclouds.
